pkg/store/prometheus.go

```diff
@@ -198,8 +198,13 @@ func (p *PrometheusStore) Series(r *storepb.SeriesRequest, s storepb.Store_Serie
 	shardMatcher := r.ShardInfo.Matcher(&p.buffers)
 	defer shardMatcher.Close()
 
+	extLsetToRemove := map[string]struct{}{}
+	for _, lbl := range r.WithoutReplicaLabels {
+		extLsetToRemove[lbl] = struct{}{}
+	}
```

Comment:
I know this is still draft, but should we remove the labels from the existing `extLset` here? This way everything else will keep working and we don't need to do `rmLabel` everywhere else.

Reply:
@fpetkovski you are correct, I had to move this chunk a bit to cover `shipChunks` case. I actually noticed that Prometheus already sends labels with external labels in them so I'm not sure why we were always calling `labels.Extend`. I have removed that.

Reply:
Makes sense. The good thing with this is that we don't need to copy labels.